src/bin/rustfmt.rs

```diff
@@ -306,8 +321,8 @@ fn determine_operation(matches: &Matches) -> FmtResult<Operation> {
             Some(dir)
         });
 
-    // if no file argument is supplied, read from stdin
-    if matches.free.is_empty() {
+    // if no file argument is supplied and `--file-lines` is not specified, read from stdin
+    if matches.free.is_empty() && !matches.opt_present("file-lines") {
```

Comment:
The reason here is that `rustfmt --file-lines='[{"file":"blah.rs", range:[1,15]}]'` with no free arguments should format `blah.rs` not stdin. I could change it so that a string like `<stdin>` is accepted in the `file` part of JSON and treated specially. This might require some validation that we're not mixing `<stdin>` and file names to avoid confusing behaviour. Thoughts?